frontend/src/components/pages/VolunteerScheduling/ConfirmShiftDetails.tsx

```diff
@@ -0,0 +1,312 @@
+import VolunteerContext from "../../../contexts/VolunteerContext";
+import { CheckIn } from "../../../types/CheckInTypes";
+import { DonorResponse } from "../../../types/DonorTypes";
+import { Schedule } from "../../../types/SchedulingTypes";
+import BackButton from "../Scheduling/BackButton";
+import { DonationSizes } from "../Scheduling/types";
+
+const schedulingDefaultData = ({
+  id: "",
+  donorId: "",
+  categories: [],
+  size: "",
+  dayPart: "",
+  startTime: "",
+  endTime: "",
+  frequency: "",
+  notes: "",
+  volunteerTime: "",
+  volunteerId: "",
+} as unknown) as Schedule;
+
+const donorDefaultData = ({
+  id: "",
+  businessName: "",
+  email: "",
+  firstName: "",
+  lastName: "",
+  phoneNumber: "",
+} as unknown) as DonorResponse;
+
+const checkInDefaultData = ({
+  id: "",
+  volunteerId: "",
+  startDate: "",
+  endDate: "",
+  notes: "",
+} as unknown) as CheckIn;
+
+const ConfirmShiftDetails = ({
+  navigation,
+  shiftId,
+  isFoodRescue,
+}: {
+  navigation: NavigationProps;
+  shiftId: string;
+  isFoodRescue: boolean;
+}) => {
+  const { previous, next } = navigation;
+  const { authenticatedUser } = useContext(AuthContext);
+  const { volunteerId } = useContext(VolunteerContext);
+  const [currentDonor, setCurrentDonor] = useState<DonorResponse>(
+    donorDefaultData,
+  );
+  const [currentFoodRescue, setCurrentFoodRescue] = useState<Schedule>(
+    schedulingDefaultData,
+  );
+  const [currentCheckIn, setCurrentCheckIn] = useState<CheckIn>(
+    checkInDefaultData,
+  );
+
+  const onSubmitClick = async () => {
+    // sign up logic
+    if (volunteerId !== null) {
+      const res = isFoodRescue
+        ? await SchedulingAPIClient.updateSchedule(shiftId, {
+          volunteerId: String(volunteerId)
+        })
+        : await CheckInAPIClient.updateCheckInById(shiftId, { 
+          volunteerId: Number(volunteerId)
+        });
+      if (!res) {
+        console.log("error when submitting volunteer");
+        return;
+      }
+    }
+    next();
+  };
+
+  const getFoodRescueResponse = async () => {
+    const foodRescueResponse = await SchedulingAPIClient.getScheduleById(
+      shiftId,
+    );
+    setCurrentFoodRescue(foodRescueResponse);
+  }
+
+  const getDonorData = async () => {
+    const donorResponse = await DonorAPIClient.getDonorById(
+      currentFoodRescue.donorId,
+    );
+    setCurrentDonor(donorResponse);
+  }
+
+  const getCheckInData = async () => {
+    const checkInResponse = await CheckInAPIClient.getCheckInsById(shiftId);
+    setCurrentCheckIn(checkInResponse);
+  };
+
+  const getDescription = () => {
+    if (currentFoodRescue.size !== "") {
+      const { description } = DonationSizes.filter(
+        (category) => category.size === currentFoodRescue.size,
+      )[0];
+      return description;
+    }
+    return "";
+  };
+
+  useEffect(() => {
+    if (isFoodRescue) {
+      getFoodRescueResponse().then(() => getDonorData());
```

Comment:
Here is where the logic is going wrong

getDonorData relies on `currentFoodRescue`  but it's being called before this is set in `useEffect`.

This does work but as discussed it probably is not best practice
```
useEffect(() => {
getDonorData()
}, [currentFoodRescue])
```

Reply:
you could combine the getting donor data and the food rescue into the same function

```typescript
getFoodRescueAndDonorResponse = async () => {
 const foodRescueResponse = await SchedulingAPIClient.getScheduleById(
      shiftId,
    );
 const donorResponse = await DonorAPIClient.getDonorById(
      currentFoodRescue.donorId,
    );
    setCurrentFoodRescue(foodRescueResponse);
    setCurrentDonor(donorResponse);
}
```
but I think this also works:
```typescript
useEffect(() => {
    (async function () {
      await getFoodRescueResponse();
      await getDonorData();
    })();
  }, []);
```

syntax might be off but yeah
the chaining you did there should also work